tests/failpoints/cases/test_rawkv.rs

```diff
@@ -164,8 +164,8 @@ fn test_leader_transfer() {
         assert_eq!(suite.must_raw_get(key1), Some(b"v4".to_vec()));
     }
 
-    // Disable CausalObserver::flush_timestamp to produce causality issue.
-    fail::cfg(FP_CAUSAL_TS_PROVIDER_FLUSH, "return").unwrap();
+    // Make causal_ts_provider.async_flush() & handle_update_max_timestamp fail.
```

Comment:
Could you explain why failpoint `FP_GET_TSO` cant make `causal_ts_provider.async_flush() & handle_update_max_timestamp` fail ?

Reply:
The critical path of `causal_ts_provider.async_flush()` is getting a batch of TSOs from PD and renew the TSO cache. So the error of `get_tso` must make `causal_ts_provider.async_flush()` fail.
Please refer to [`BatchTsoProvider::renew_tso_batch_impl()`](https://github.com/tikv/tikv/blob/1a9446f334c2d29417f656fa8ef4ec6a1dda95f7/components/causal_ts/src/tso.rs#L419).

And `handle_update_max_timestamp()` invokes `causal_ts_provider.async_flus()` [here](https://github.com/tikv/tikv/blob/1a9446f334c2d29417f656fa8ef4ec6a1dda95f7/components/raftstore/src/store/worker/pd.rs#L1622) (when `storage.api-version = 2`). So `handle_update_max_timestamp()` will also fail (and keep retrying until succeed).